### PR TITLE
[SYCL][Doc] Add isnan to bfloat16 math doc

### PR DESCRIPTION
**Pull request by jinge90**
Signed-off-by: jinge90 <ge.jin@intel.com>

### CODE REVIEW

**Comment by jinge90** [comment]
Hi, @gmlueck and @JackAKirk 
I added "isnan" for bfloat16 in https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/oneapi/experimental/bfloat16_math.hpp#L36
This PR adds it in bfloat16 math doc. However, I didn't add "isnan" for sycl::marray<sycl::ext::oneapi::bfloat16, N> and wanted to consult you what the prototype should look like?
For bfloat16, isnan prototype is:
bool isnan(sycl::ext::oneapi::bfloat16);
For sycl::marray<sycl::ext::oneapi::bfloat16, N>, I expect the prototype to look like:
sycl::marray<bool, N> isnan(sycl::marray<sycl::ext::oneapi::bfloat16, N>);
Each bool value in return value corresponds each bf16 value in input param.

In CUDA math, NV has "hisnan2": https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH____BFLOAT162__COMPARISON.html#group__CUDA__MATH____BFLOAT162__COMPARISON_1gf613f74f4c0220e6d85b91b086e2788a
hisnan2 return turn value is a __nv_bfloat162 with the corresponding nv_bfloat16 results set to 1.0 for NaN, 0.0 otherwise.

So, what's your idea? Do we need to align with NV math? If so, the return type should be sycl::marray<sycl::ext::oneapi::bfloat16, N>.

Thanks very much.

**Comment by JackAKirk** [comment]
> Hi, @gmlueck and @JackAKirk I added "isnan" for bfloat16 in https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/oneapi/experimental/bfloat16_math.hpp#L36 This PR adds it in bfloat16 math doc. However, I didn't add "isnan" for sycl::marray<sycl::ext::oneapi::bfloat16, N> and wanted to consult you what the prototype should look like? For bfloat16, isnan prototype is: bool isnan(sycl::ext::oneapi::bfloat16); For sycl::marray<sycl::ext::oneapi::bfloat16, N>, I expect the prototype to look like: sycl::marray<bool, N> isnan(sycl::marray<sycl::ext::oneapi::bfloat16, N>); Each bool value in return value corresponds each bf16 value in input param.
> 
> In CUDA math, NV has "hisnan2": https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH____BFLOAT162__COMPARISON.html#group__CUDA__MATH____BFLOAT162__COMPARISON_1gf613f74f4c0220e6d85b91b086e2788a hisnan2 return turn value is a __nv_bfloat162 with the corresponding nv_bfloat16 results set to 1.0 for NaN, 0.0 otherwise.
> 
> So, what's your idea? Do we need to align with NV math? If so, the return type should be sycl::marray<sycl::ext::oneapi::bfloat16, N>.
> 
> Thanks very much.

Hmm well spotted. My guess is they have only done this because they don't have a corresponding __nv_bool2 to __nv_bfloat162. My instinct is to agree with you that in the SYCL case it should return sycl::marray<bool, N>.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_bfloat16_math_functions.asciidoc [hunk 1]
```diff
@@ -104,6 +104,22 @@ The following functions are only available when `T` is `bfloat16` or
 `sycl::marray<bfloat16, {N}>`, where `{N}` means any positive value of
 `size_t` type.
 
+==== isnan
+
+```c++
+namespace sycl::ext::oneapi::experimental {
+
+template <typename T>
+bool isnan(T x);
```
Comment (JackAKirk) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_bfloat16_math_functions.asciidoc:113]: A space would be good here for better clarity.

```suggestion
bool isnan(T x);

Template <size_t N>
```

**Comment by JackAKirk** [comment]
LGTM

**Review by JackAKirk — APPROVED** [review]

**Comment by jinge90** [comment]
Hi, @intel/dpcpp-specification-reviewers and @gmlueck 
Kind ping~
Thanks very much.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_bfloat16_math_functions.asciidoc [hunk 2]
```diff
@@ -104,6 +104,23 @@ The following functions are only available when `T` is `bfloat16` or
...
```
Comment (gmlueck) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_bfloat16_math_functions.asciidoc:113]:
```suggestion
bool isnan(bfloat16 x);
```

The paragraph above says that `T` must either be `bfloat16` or `sycl::marray<bfloat16, {N}>`.  For `isnan`, there are different overloads for these two cases, so I don't think you want to use a template with `T` here.

**Comment by jinge90** [comment]
Hi, @gmlueck 
Is there anything else in this PR which needs update?

Thanks very much.

**Review by gmlueck — APPROVED** [review]

**Comment by gmlueck** [comment]
> Is there anything else in this PR which needs update?

No, LGTM!

**Pull request merged**